addons/sale/wizard/payment_link_wizard.py

```diff
@@ -17,7 +17,7 @@ def default_get(self, fields):
             record = self.env[res['res_model']].browse(res['res_id'])
             res.update({
                 'description': record.name,
-                'amount': record.amount_total - sum(record.invoice_ids.filtered(lambda x: x.state != 'cancel').mapped('amount_total')),
+                'amount': record.amount_total - sum(record.invoice_ids.filtered(lambda x: x.state != 'cancel' and x.invoice_origin == record.name).mapped('amount_total')),
```

Comment:
I don't know if this `invoice_origin` mechanism is robust enough. These values are always correct on real database (at least for recent records) ?We could use the `x.invoice_line_ids.sale_line_ids.order_id == record` but it will trigger more lookup in the tables. I let you decide what you think is best approach :-).

Reply:
@arj-odoo I totally agree, I also wasn't convinced about using `invoice_origin`, using an `id` is better ! thank you for the feedback :pray:

Reply:
@arj-odoo can you please R+ the fix if you thinks everything is ok ?